apps/hash-graph/lib/graph/src/store/query.rs

```diff
@@ -13,6 +18,39 @@ pub trait QueryPath {
     fn expected_type(&self) -> ParameterType;
 }
 
+/// Parses a query token of the form `token(key=value)`.
+///
+/// Whitespaces are ignored and multiple parameters are supported.
+///
+/// # Errors
+///
+/// - If the token is not of the form `token`, `token()`, or `token(key=value)`
+/// - If `token` can not be deserialized into `T`
+pub fn parse_query_token<'de, T: Deserialize<'de>, E: de::Error>(
+    token: &'de str,
+) -> Result<(T, HashMap<&str, &str>), E> {
+    if let Some((token, parameters)) = token.split_once('(') {
+        let parameters = parameters
+            .strip_suffix(')')
+            .ok_or_else(|| E::custom("missing closing parenthesis"))?;
+
+        let parameters = parameters
+            .split(',')
+            .filter(|parameter| !parameter.trim().is_empty())
```

Comment:
Required for `"".split(',')` resulting in `[""]`. That `",,"` will be allowed here as well is a side effect but also not really an issue.